### DISCUSSION TITLE
Issues with connecting machine

### DISCUSSION THREAD

**Discussion by BlockheadVFX-IT** [rustdesk/rustdesk-server-pro#224] [General]
Hi All, 
We have been using RustDesk Pro in our enterprise environment since last couple of months. We have 2 offices across Australia (Sydney) and New Zealand (Auckland). The primary id server is in Auckland side and the extra relay server is in Sydney (standalone hbbr free version). 

Lets give it a name so it will be better to relate 

AKL08VRUSTDESK01= Main ID Server , and its a Virtual Machine , Ubuntu Server, ports 21114,21115,21116 forwarded to this RustDesk Server 

AKL08VHBBR01 == standalone hbbr (relay server) , and ,  21117 and 21119 port forwarded to this docker container for external users 

SYD02VHBBR01 = Relay server in Sydney for both internal and external users, 21117 and 21119 port forwarded to this docker container 

Upon checking the logs and stuff , i can confirm that connection requests are being sent to Sydney relay server for Sydney devices and some requests for Sydney are being pointed to main Auckland server, even though i initiate connection from Sydney. 
Shouldn't the closest server be used i.e. SYD02HBBR01?

Or lets say, while initiating the connection , lets say it went to AKL08VRUSTDESK01, but shouldn't hbbs server dictate the client or at least negotiate with Sydney relay server and get the client connected to SYD02VHBBR01 to make a connection to remote workstation located in sydney?

FYI , this experiment was particularly done from my home to office, so client----remote machines are in different network.

Other  issue  is when trying to establish the connection, we have to hit reconnect for multiple times (sometimes 5-8 time) before connecting to the remote workstation. This issue happens both internally and externally. What might be causing this issue?

We have decent specs allocated for both hbbr and hbbs server and Internet speed is 2 Gbps in Sydney Office and Similar in Auckland 

Can someone help me to understand this issue?

**Comment by BlockheadVFX-IT** [comment]
All relay servers work , but i am getting very mixed experience (while establishing connection)

So basically, how can we make AKL08VRUSTDESK01 smart enough to redirect to nearest location. FYI i have also configured the geo location database as per rustdesk's documentation
![image](https://github.com/rustdesk/rustdesk-server-pro/assets/152861034/9660d169-28f4-4632-bb10-508d8c393310)

**Reply by rustdesk** [reply]
What pro version are you using? I did found a bug about this.

**Reply by BlockheadVFX-IT** [reply]
1.2.0

**Reply by rustdesk** [reply]
1.2.0 has included the fix of the bug

**Reply by rustdesk** [reply]
You can try out the "Get Relay Server" button.

![image](https://github.com/rustdesk/rustdesk-server-pro/assets/71636191/19edccd1-9dd4-454a-bb18-ce66fda7a7c8)

**Reply by rustdesk** [reply]
To cofirm the GEO used for relay servers, you can run `echo rs | netcat 127.0.0.1 21115`, `21115` is the port of `hbbs`.

`relay-server-ip Some(long *10, lat * 10)`

![image](https://github.com/rustdesk/rustdesk-server-pro/assets/71636191/41b647e4-6ea3-4b9e-870b-d095e15e8501)

**Reply by SagarKandel** [reply]
Seems to have fixed the issue. Used both public and private ip address of relay servers and  maybe that's what confused the signalling server. 
Only using public ip addresses of Sydney and well as Auckland office made everything stable. 

Man o man, its blazing fast. 

Thanks @rustdesk 

Also we'd like to see if shared address book (just how TeamViewer has) in near future, so that it will be easier to collaborate with different departments , letting them have access to the only machines assigned to p[articular group.

**Reply by rustdesk** [reply]
Shared address book is almost ready for release.

**Reply by rustdesk** [reply]
https://twitter.com/rustdesk/status/1777180459429986787
